src/wallet/coinselection.h

```diff
@@ -73,16 +75,49 @@ struct COutput {
           solvable{solvable},
           safe{safe},
           time{time},
-          from_me{from_me},
-          effective_value{txout.nValue}
-    {}
+          from_me{from_me}
+    {
+        if (fees) {
+            // if input_bytes unknown, the fees should be 0, if input_bytes is known, then the fees should be a positive integer
+            assert((input_bytes < 0 && fees.value() == 0) || (input_bytes > 0 && fees.value() >= 0));
+            fee = fees;
+            effective_value = txout.nValue - fee.value();
+        }
+    }
+
+    COutput(const COutPoint& outpoint, const CTxOut& txout, int depth, int input_bytes, bool spendable, bool solvable, bool safe, int64_t time, bool from_me, const CFeeRate feerate)
```

Comment:
I think it would be better to have `feerate` be a `std::optional<CFeeRate` with a default of `std::nullopt`. This would allow us to do some simplification in `AvailableCoins` where the `CFeeRate` can just be passed in without checking whether it has a value.

Reply:
I agree, it looks a lot cleaner now.